packages/cli/src/services/orchestration.service.ts

```diff
@@ -44,7 +44,8 @@ export class OrchestrationService {
 	}
 
 	/**
-	 * Whether this instance is the leader in a multi-main setup. Always `false` in single-main setup.
+	 * Whether this instance is the leader. This is always `true` in single-main setup,
+	 * once `OrchestrationService` has been initialized.
```

Comment:
`WaitTracker` is initialized as part of `ExecutionService`, which is initialized as part of `ActiveWorkflowRunner`, which is initialized before `OrchestrationService` is initialized. This is why at #9100 `WaitTracker.constructor` found `isLeader` to be `false` in single-main setup - because `instanceType` is `unset` until `OrchestrationService` is initialized. 

Ideally `OrchestrationService` should be initialized before `ActiveWorkflowRunner` but this original order has been in place for a long time - changing this may have unforeseen side effects, so I decided to preserve the order of `OrchestrationService` and `ActiveWorkflowRunner` for now.